src/thor/triplegbuilder_utils.h

```diff
@@ -339,4 +339,172 @@ struct MultimodalBuilder {
   }
 };
 
+/**
+ * Add trip intersecting edge.
+ * @param  controller   Controller to determine which attributes to set.
+ * @param  directededge Directed edge on the path.
+ * @param  prev_de  Previous directed edge on the path.
+ * @param  local_edge_index  Index of the local intersecting path edge at intersection.
+ * @param  nodeinfo  Node information of the intersection.
+ * @param  trip_node  Trip node that will store the intersecting edge information.
+ * @param  intersecting_de Intersecting directed edge. Will be nullptr except when
+ *                         on the local hierarchy.
+ */
+void AddTripIntersectingEdge(const AttributesController& controller,
```

Comment:
i removed this from `triplegbuilder::build` because its huge and distracting. the hope is that one day that function will be lean and mean and a normal human will be able to understand it